From: Thomas Zajic <zlatko-github@zlatk0.net>
Date: Wed, 17 Nov 2021 17:12:04 +0000
Subject: Flush drawable surface when getting a pixbuf

When getting the drawable surface to get its pixels we should be
flushing the Cairo state, not mark it dirty.

Bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/4456
Origin: upstream, 3.24.36, commit:e413f5c43259a22269aa1e75767dd545e10119b5
---
 gdk/gdkpixbuf-drawable.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/gdkpixbuf-drawable.c b/gdk/gdkpixbuf-drawable.c
index 603c8ff..e63c3a8 100644
--- a/gdk/gdkpixbuf-drawable.c
+++ b/gdk/gdkpixbuf-drawable.c
@@ -104,7 +104,7 @@ gdk_pixbuf_get_from_window (GdkWindow *src,
    * by external applications.
    * So be on the safe side and:
    */
-  cairo_surface_mark_dirty (surface);
+  cairo_surface_flush (surface);
 
   if (cairo_surface_get_content (surface) & CAIRO_CONTENT_ALPHA)
     copy = cairo_image_surface_create (CAIRO_FORMAT_ARGB32, width * scale, height * scale);
